src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,93 @@
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+
+import TreeListItem from './TreeListItem'
+import { UIComponent, childrenExist, customPropTypes, createShorthandFactory } from '../../lib'
+import { ComponentSlotStyle, ComponentVariablesInput } from '../../themes/types'
+import { ComponentEventHandler, ShorthandValue } from '../../../types/utils'
+import { treeBehavior } from '../../lib/accessibility'
+import { Accessibility } from '../../lib/accessibility/types'
+
+export type TreeProps = {
+  as?: any
+  children?: React.ReactChildren
+  content?: React.ReactNode
+  styles?: ComponentSlotStyle
+  variables?: ComponentVariablesInput
+  treedata: {
+    title: ShorthandValue
+    onItemClick?: ComponentEventHandler<TreeProps>
+    subtree?: any[]
+  }[]
+  isSubTree?: boolean
+}
+
+class Tree extends UIComponent<TreeProps, any> {
+  static create: Function
+
+  static className = 'tree-list'
+
+  static displayName = 'Tree'
+
+  // static handledProps = ['as', 'children', 'content', 'styles', 'variables']
+
+  static propTypes = {
+    /** An element type to render as. */
+    as: customPropTypes.as,
+
+    /** Define your own children. */
+    children: PropTypes.node,
+
+    /** Shorthand for primary content. */
+    content: PropTypes.any,
+
+    /** Custom styles to be applied to the component. */
+    styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** Custom variables to be applied to the component. */
+    variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** Shorthand array of props for Tree. */
+    treedata: PropTypes.array,
+
+    /** Whether the tree is a subtree. */
+    isSubTree: PropTypes.boolean,
+  }
+
+  public static defaultProps = {
+    as: 'ul',
+    accessibility: treeBehavior as Accessibility,
+  }
+
+  renderContent(styles, variables) {
+    const { treedata } = this.props
+    if (!treedata) return []
+    return treedata.map(obj => {
```

Comment:
We can use there map from `lodash`, it will be useless condition above:

```suggestion
    return _.map(items, item => {
```